### ISSUE TITLE
Splash Screen

### ISSUE THREAD

**Issue by dinurymomshad** [KalleHallden/exer_log#68]
If we have a splash screen then we can check
1. If the user is logged in
2. If the app is newly installed
3. This would also solve the black screen issue, after logging in.

**From the design side, I am thinking**: A screen with Logo and App Name in the middle. The background color is our app's primary background color.

**Comment by 0RaMsY0** [comment]
this a function i use for the splash screen
``` dart
import 'package:flutter/material.dart';
import 'package:readlex/main.dart';

class SplashScreen extends StatefulWidget {
  Widget whatToReturn;  // we are gonna get a widget to return in these case it will be the HomePage
  SplashScreen({Key? key, required Widget this.whatToReturn}) : super(key: key);

  @override
  State<SplashScreen> createState() => _SplashScreenState();
}

class _SplashScreenState extends State<SplashScreen> {
  @override
  void initState() {
    super.initState();
    _navigateToTheMainPage();
  }

  _navigateToTheMainPage() async {
      // we are making a delay for 3 seconds then navigating to the widget that got pass to the whatToReturn variable
    await Future.delayed(Duration(seconds: 3), () {
      Navigator.pushReplacement(
          context, MaterialPageRoute(builder: (context) => widget.whatToReturn));
    });
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      // your spalsh screen content 
    );
  }
}

```
this is how i use it in my project
the ShowPage widget is what i call as the home in the MaterialApp btw
``` dart
class ShowPage extends StatelessWidget {
  const ShowPage({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return StreamBuilder<User?>(
        stream: FirebaseAuth.instance.authStateChanges(),
        builder: (context, snapshot) {
          if (snapshot.hasData) {
            return SplashScreen(whatToReturn: HomePage()); //return the splash screen that make a delay for 3 seconds then return the HomePage() in case of been login in
          } else {
            return LoginPage(); // return the login screen in case of not been login in
          }
        });
  }
}
```

pls if i have a mistake or something that well help emprove this then tell me i'am still a beginner at flutter,

**Comment by dinurymomshad** [comment]
@0RaMsY0  This is mostly similar to what I have in mind. But instead of stream builder, we will rely on providers from riverpod. I will work on this today. You can check my implementation and let me know if you have suggestions.

**Comment by suzanpradhan** [comment]
I guess it's better to stick with native splash rather than making our own in flutter. The native splash will be displayed from start app until calling the `runApp()` and you can initialize Firebase part before `runApp`.

**Comment by ramsy0dev** [comment]
@dinurymomshad ok

**Comment by momshaddinury** [comment]
@suzanpradhan but we need to customize the UI. Are you suggesting we customize the native UI?

**Comment by suzanpradhan** [comment]
There are already many packages that will do it for us. Popular one might be: https://pub.dev/packages/flutter_native_splash

**Comment by ramsy0dev** [comment]
@suzanpradhan i think it better to hardcode it then using a third-party package

**Comment by suzanpradhan** [comment]
Yeah, you are right and you will have more customization too like adding animations in splash screen. But, since we are having only logo and name, it's fine to let package do it instead of doing manual work.

**Comment by ramsy0dev** [comment]
@suzanpradhan  Ok i see what you mean

**Comment by jorre127** [comment]
Personally I would actually use the splashscreen in flutter. This would give us more flexibility in customising the splash screen (adding animations and such). It would also allow us to be able to put all the initialization logic (firebase, getting data from database/api, etc.) there, instead of putting all this before the runApp, which isn't really desirable imo. If we make the flutter splash look the same as the native one, the users wouldn't even notice the transition.

**Comment by suzanpradhan** [comment]
Hmm, putting all the initializations before `runApp()` could be tedious. Then, lets create the same splash screen as native one. Native splash will be displayed on app bootup then flutter one is shown during initialization.

**Comment by KalleHallden** [comment]
I believe #85 now closes this issue right?

**Comment by momshaddinury** [comment]
> I believe #85 now closes this issue right?

Yes. I am not able to link issues to PR for some reason. Is that normal? I think only people with write-access can do that.

**Comment by Ki-er** [comment]
> > I believe #85 now closes this issue right?
> 
> Yes. I am not able to link issues to PR for some reason. Is that normal? I think only people with write-access can do that.



https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/autolinked-references-and-urls

#85

Anyone can link issues by doing #<issue number>

If you just want to reference it then you can do Ref #85, or if it closes it then you can do `Closes #85`

**Comment by momshaddinury** [comment]
Nah that is not working. I will close this now.
